applications/crossbar/src/modules/cb_transactions.erl

```diff
@@ -111,29 +181,102 @@ validate_transactions(Context, ?HTTP_GET) ->
-validate_transaction(Context, <<"monthly_recurring">>, ?HTTP_GET) ->
+validate_transaction(Context, ?MONTHLY, ?HTTP_GET) ->
     case cb_modules_util:range_view_options(Context) of
         {CreatedFrom, CreatedTo} ->
             Reason = cb_context:req_value(Context, <<"reason">>),
             fetch_monthly_recurring(Context, CreatedFrom, CreatedTo, Reason);
         Context1 -> Context1
     end;
-validate_transaction(Context, <<"subscriptions">>, ?HTTP_GET) ->
+validate_transaction(Context, ?SUBSCRIPTIONS, ?HTTP_GET) ->
     filter_subscriptions(Context);
+validate_transaction(Context, ?DEBIT, ?HTTP_DELETE) ->
+    validate_debit(Context);
 validate_transaction(Context, _PathToken, _Verb) ->
     cb_context:add_system_error('bad_identifier',  Context).
 
 
+%%--------------------------------------------------------------------
 %% @private
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec validate_debit(cb_context:context()) -> cb_context:context().
+-spec validate_debit(cb_context:context(), 'undefined' | float()) -> cb_context:context().
+validate_debit(Context) ->
+    JObj = cb_context:req_data(Context),
+    Amount = wh_json:get_float_value(<<"amount">>, JObj),
+
+    AuthAccountId = cb_context:auth_account_id(Context),
+
+    case cb_modules_util:is_superduper_admin(Context) of
+        'true' -> validate_debit(Context, Amount);
+        'false' ->
+            case wh_services:is_reseller(AuthAccountId) of
+                'true' -> validate_debit(Context, Amount);
+                'false' -> cb_context:add_system_error('forbidden', Context)
+            end
+    end.
+
+validate_debit(Context, 'undefined') ->
+    Message = <<"Amount is required">>,
+    cb_context:add_validation_error(
+        <<"amount">>
+        ,<<"required">>
+        ,wh_json:from_list(
+            [{<<"message">>, Message}]
+        )
+        ,Context
+    );
+validate_debit(Context, Amount) when Amount =< 0 ->
+    Message = <<"Amount is must be more than $0">>,
+    cb_context:add_validation_error(
+        <<"amount">>
+        ,<<"minimum">>
+        ,wh_json:from_list(
+            [{<<"message">>, Message}]
+        )
+        ,Context
+    );
+validate_debit(Context, Amount) ->
+    AccountId = cb_context:account_id(Context),
+    FuturAmount = current_account_dollars(AccountId) - Amount,
+    case FuturAmount < 0 of
+        'false' ->
+            cb_context:set_resp_status(Context, 'success');
+        'true' ->
+            Message = <<"Available credit can not be less than $0">>,
+            cb_context:add_validation_error(
+                <<"amount">>
+                ,<<"minimum">>
+                ,wh_json:from_list([
+                    {<<"message">>, Message}
+                    ,{<<"cause">>, FuturAmount}
+                ])
+                ,Context
+            )
+    end.
+
+-spec current_account_dollars(ne_binary()) -> float().
+current_account_dollars(Account) ->
```

Comment:
I think this function is in the teletype-related templates too. Perhaps `wht_util` should be the home for this function?